app/assets/javascripts/discourse/templates/modal/edit-category.hbs

```diff
@@ -12,7 +12,7 @@
 
   {{#d-modal-body}}
     {{#each panels as |tab|}}
-      {{component tab selectedTab=selectedTab category=model}}
+      {{component tab selectedTab=selectedTab category=model modal=this}}
```

Comment:
You shouldn't pass the component/controller down into another component like this. I would suggest instead passing an action that can be called by the component.

Reply:
Ok I had to remove the panels loop because the rest of the tabs don't need the action and the extra attributes. Let me know what you think.

Reply:
Never mind this^^, I've simplified the code which also removed the need for an observer or delay. Let me what you think.